flowcraft/templates/run_mlst.py

```diff
@@ -0,0 +1,431 @@
+    """
+    Parses the mlst scheme database and returns the full scheme for the species
+    and the genus.
+
+    Parameters
+    ----------
+    species_splited: list
+        List with the genus and specific epithet for the expected species
+    mlst_db_path: str
+        Path to the mlst database containing the species and respective schema
+    species_scheme_map_version: int
+        Version on the mlst database
+
+    Returns
+    -------
+    scheme: str
+        mlst scheme name
+    genus_mlst_scheme: str
+        genus name for the mlst scheme
+    """
+    scheme = 'unknown'
+    genus_mlst_scheme = None
+
+    with open(mlst_db_path, 'rtU') as reader:
+        for line in reader:
+            line = line.splitlines()[0]
+
+            if len(line) > 0:
+
+                if not line.startswith('#'):
+                    line = line.lower().split('\\t')
+                    line = [line[i].split(' ')[0] for i in range(0, len(line))]
+                    val_genus, val_species, val_scheme = set_species_scheme_map_variables(line, species_scheme_map_version)
+
+                    if val_genus == species_splited[0]:
+
+                        if val_species == '':
+                            genus_mlst_scheme = val_scheme
+
+                        elif val_species == species_splited[1]:
+                            scheme = val_scheme
+
+        if scheme == 'unknown' and genus_mlst_scheme is not None:
+            scheme = genus_mlst_scheme
+
+    return scheme, genus_mlst_scheme
+
+
+def clean_novel_alleles(novel_alleles, scheme_mlst, profile):
+    """
+    Clean the fasta file with the novel alleles produced by mlst
+
+    Parameters
+    ----------
+    novel_alleles : str
+        Path for fasta file containing the novel alleles
+    scheme_mlst : str
+        MLST schema found by mlst
+    profile : list
+        List of strings with the profile found
+
+    Returns
+    -------
+    """
+    unknown_genes = []
+    for gene_allele in profile:
+        gene = gene_allele.split('(')[0]
+        try:
+            allele = gene_allele.split('(')[1].rstrip(')')
+            if allele.startswith('~'):
+                unknown_genes.append(gene)
+        except IndexError as e:
+            print('WARNING: {}'.format(e))
+
+    try:
+        novel_alleles_keep = {}
+        if len(unknown_genes) > 0:
+            reader = open(novel_alleles, mode='rt', newline=None)
+            fasta_iter = (g for k, g in groupby(reader, lambda x: x.startswith('>')))
+            for header in fasta_iter:
+                header = header.__next__()[1:].rstrip('\\r\\n')
+                seq = ''.join(s.rstrip('\\r\\n') for s in fasta_iter.__next__())
+                if header.startswith(scheme_mlst):
+                    gene = header.split('.')[1].split('~')[0]
+                    if gene in unknown_genes:
+                        novel_alleles_keep[header] = seq
+            reader.close()
+
+        os.remove(novel_alleles)
```

Comment:
Is this intended to remove the file? Wouldn't this file be stored in work dir and then if the user wants remove it? Or is it too big and this is for saving disk space?